ui/pages/confirmations/hooks/useConfirmationAlertMetrics.ts

```diff
@@ -0,0 +1,132 @@
+import { TransactionType } from '@metamask/transaction-controller';
+import { useCallback, useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { validate as isUuid } from 'uuid';
+import useAlerts from '../../../hooks/useAlerts';
+import { REDESIGN_TRANSACTION_TYPES } from '../utils';
+import { Alert } from '../../../ducks/confirm-alerts/confirm-alerts';
+import { confirmSelector } from '../../../selectors';
+import { AlertsName } from './alerts/constants';
+import { useTransactionEventFragment } from './useTransactionEventFragment';
+
+export type UseAlertSystemMetricsProps = {
+  alertKey?: string;
+  action?: AlertsActionMetrics;
+};
+
+export type AlertMetricsProperties = {
+  alert_visualized: string[];
+  alert_visualized_count: number;
+  alert_key_clicked: string[];
+  alert_action_clicked: string[];
+};
+
+export const ALERTS_NAME_METRICS: Record<AlertsName | string, string> = {
+  [AlertsName.GasEstimateFailed]: 'gas_estimate_failed',
+  [AlertsName.GasFeeLow]: 'gas_fee_low',
+  [AlertsName.GasTooLow]: 'gas_too_low',
+  [AlertsName.InsufficientBalance]: 'insufficient_balance',
+  [AlertsName.NetworkBusy]: 'network_busy',
+  [AlertsName.NoGasPrice]: 'no_gas_price',
+  [AlertsName.PendingTransaction]: 'pending_transaction',
+  [AlertsName.SigningOrSubmitting]: 'signing_or_submitting',
+  [AlertsName.Blockaid]: 'blockaid',
+};
+
+export enum AlertsActionMetrics {
+  AlertVisualized = 'AlertVisualized',
+  InlineAlertClicked = 'InlineAlertClicked',
+  AlertActionClicked = 'AlertActionClicked',
+}
+
+export function useConfirmationAlertMetrics() {
+  const { currentConfirmation } = useSelector(confirmSelector);
+  const ownerId = currentConfirmation?.id ?? '';
+  const { alerts, isAlertConfirmed } = useAlerts(ownerId);
+  const { updateTransactionEventFragment } = useTransactionEventFragment();
+
+  const [metricsProperties, setMetricsProperties] =
+    useState<AlertMetricsProperties>({
+      alert_visualized: [],
+      alert_visualized_count: 0,
+      alert_key_clicked: [],
+      alert_action_clicked: [],
+    });
+
+  // Temporary measure to track metrics only for redesign transaction types
+  const isValidType = REDESIGN_TRANSACTION_TYPES.includes(
+    currentConfirmation?.type as TransactionType,
+  );
+
+  const trackAlertMetrics = useCallback(
+    (props?: UseAlertSystemMetricsProps) => {
+      const { alertKey, action } = props ?? {};
+      if (!isValidType || !alertKey || !action) {
+        return;
+      }
+
+      setMetricsProperties((prevState) => {
+        const newState = { ...prevState };
+        const alertName = isUuid(alertKey)
```

Comment:
As this logic is also in `getAlertNames`, could we move it to a tiny `getAlertName` function?